src/ripple/protocol/impl/Feature.cpp

```diff
@@ -20,17 +20,43 @@
 #include <BeastConfig.h>
 #include <ripple/protocol/digest.h>
 #include <ripple/protocol/Feature.h>
+
+#include <boost/container/flat_map.hpp>
+
 #include <cstring>
 
 namespace ripple {
 
+static boost::container::flat_map<uint256, std::size_t> featureToIndex;
+static std::vector<uint256> features;
```

Comment:
Although the `FeatureBitset` is only being used by `Env` at the moment, there's the potential for use elsewhere in the server.  I think that means we should put some thought into thread protection when these data structures are being constructed.